pkg/trace/obfuscate/sql_test.go

```diff
@@ -83,6 +83,30 @@ func TestKeepSQLAlias(t *testing.T) {
 	})
 }
 
+func TestCanObfuscateAutoVacuum(t *testing.T) {
+	assert := assert.New(t)
+	for _, tt := range []struct{ in, out string }{
+		{
+			in:  "autovacuum: VACUUM ANALYZE public.monitor",
+			out: "autovacuum :  VACUUM ANALYZE public.monitor",
```

Comment:
OK, @justiniso it looks like I can't get around this very easily. Here is the flow of the current code:

1. We get to the `:` char and call [`tokenizer.Scan`](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/obfuscate/sql.go#L344)
2. We hit [this case](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/obfuscate/sql_tokenizer.go#L221-L229), and then (with my fix) we would return the current token, which is `:`
3. We come out of the Scan func, and then [hit this code](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/obfuscate/sql.go#L366-L379), which will write an extra `'  '` char after the `:`. Since we do not have have access the **next token** we can't know if this is something we should add a space to or not. 

I think we can "fix" this by creating a special `TokenKind` (like `Vacuum`? or _something_), and then add a [case here](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/obfuscate/sql.go#L368) to skip the space if we see this token. 🤷  seems kinda meh. 

More limitations of using the token parser I guess

Reply:
Talked about this offline, and decided we are OK with this behavior for now

Reply:
Hey @jmeunier28 after considering some of the edge cases like `@runtot:=0` in the tests, we have to do one of two things:

1. apply grammar in the tokenizer (conditional logic that looks at the next token and checks if it is a space). This would be really bad and could introduce a lot of nasty bugs.
2. add a rule in our "lexer" (i.e. the `tokenFilter` in `sql.go`) to concat previous tokens to single `:`. This is less bad, but more complicated and requires expanding the functionality of the lexer a bit more to support backtracking or peeking forward.

I think this case is small enough that it won't be an issue. We can apply #2 later, but the important thing is fixing the error so these queries properly obfuscate.